src/parser/type_checking.c

```diff
@@ -22979,11 +22980,18 @@ pt_common_collation (PT_COLL_INFER * arg1_coll_infer, PT_COLL_INFER * arg2_coll_
 	{
 	  goto error;
 	}
-      *common_coll = arg2_coll_infer->coll_id;
+      if (arg2_coll_infer->coll_id > arg1_coll_infer->coll_id)
+	{
+	  *common_coll = arg1_coll_infer->coll_id;
+	}
+      else
+	{
+	  *common_coll = arg2_coll_infer->coll_id;
+	}
       *common_cs = arg2_coll_infer->codeset;
 
       /* check arg3 */
-      if (op_has_3_args && arg3_coll_infer->coll_id != *common_coll)
+      if (op_has_3_args && !IS_SAME_VARIATION (arg3_coll_infer->coll_id, *common_coll))
```

Comment:
Why should TS and TI be considered the same collation?

Reply:
if it's not checked, it cause the collation compatibility fail.
for example, select * from t1, t2 where t1.a = t2.a; which t1.a is ti-collation and t2.a is ts-collation.